contracts/javascore/ibc/src/main/java/ibc/ics20/app/ICS20Transfer.java

```diff
@@ -0,0 +1,179 @@
+package ibc.ics20.app;
+
+import ibc.icon.interfaces.IIBCModule;
+import ibc.icon.score.util.Logger;
+import ibc.icon.score.util.StringUtil;
+import ibc.ics05.port.ModuleManager;
+import ibc.ics25.handler.IBCHandler;
+
+import icon.proto.core.channel.Channel;
+import icon.proto.core.channel.Packet;
+import score.Address;
+import score.Context;
+import score.DictDB;
+import score.annotation.External;
+
+import java.math.BigInteger;
+import java.util.Map;
+
+public class ICS20Transfer {
+    public static String ICS20_VERSION = "ics20-1";
+    public static Address ZERO_ADDRESS = Address.fromString("hx0000000000000000000000000000000000000000");
+
+
+    public static final DictDB<String, Address> channelEscrowAddresses = Context.newDictDB("channelEscrowAddresses", Address.class);
+
+
+    @External
+    public byte[] onRecvPacket(byte[] packet, Address relayer) {
+//        TODO unmarshal json
+        boolean success = false;
```

Comment:
validate amount send in packet to not be zero  as well.

Reply:
checked